### PR TITLE
Rename argnum with argnums in Catalyst grad related functions

### PR DESCRIPTION
**Pull request by rmoyard**
**Context:**
Catalyst is renaming argnum with argnums. https://github.com/PennyLaneAI/catalyst/pull/1036

**Description of the Change:**
Update calls to Catalyst functions.

### CODE REVIEW

**Review by josh146 — APPROVED** [review]
Thanks @rmoyard! Don't forget to update the changelog.

Anything we need to keep in mind here, regarding:

- minimum catalyst version in the PL tests
- order in which we merge the PRs
- things the release managers should know?

**Comment by rmoyard** [comment]
@josh146 

- I will update the Catalyst version to use the one from test pypi of tomorrow
- I will merge the Catalyst one first and then this one because we need to install Catalyst
- Tag @mudit2812 and @rauletorresc for visibility

File: .github/workflows/install_deps/action.yml [hunk 1]
```diff
@@ -98,7 +98,8 @@ runs:
     - name: Install Catalyst
       shell: bash
       if: inputs.install_catalyst_after_pennylane == 'true'
-      run: pip install --upgrade pennylane-catalyst
+      # TODO: replace after release
+      run: pip install --extra-index-url https://test.pypi.org/simple/ pennylane-catalyst --pre
```
Comment (rmoyard) [inline .github/workflows/install_deps/action.yml:102]: TODO: add version after https://github.com/PennyLaneAI/pennylane/pull/6117 is merged and wheels are on test pypi

**Review by lillian542 — APPROVED** [review]
Looks good to me!

**Pull request merged**